bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/wrapper/VehiclePositionWrapper.java

```diff
@@ -81,11 +77,11 @@ public State isHeading() {
     }
 
     public State getTimestamp() {
-        return vehicle.position.timestamp != null
-                ? new DateTimeType(vehicle.position.timestamp.withZoneSameInstant(ZoneId.systemDefault()))
-                : vehicle.calculatedPosition.timestamp != null
-                        ? new DateTimeType(
-                                vehicle.calculatedPosition.timestamp.withZoneSameInstant(ZoneId.systemDefault()))
+        return vehicle.position.getTimestamp().isPresent()
+                ? new DateTimeType(vehicle.position.getTimestamp().get().withZoneSameInstant(ZoneId.systemDefault()))
+                : vehicle.calculatedPosition.getTimestamp().isPresent()
+                        ? new DateTimeType(vehicle.calculatedPosition.getTimestamp().get()
+                                .withZoneSameInstant(ZoneId.systemDefault()))
                         : UnDefType.NULL;
```

Comment:
This is really hard to read. Since we only have JDK8 available, maybe something like this

```
ZonedDateTime dt = vehicle.position.getTimestamp().orElse(vehicle.calculatedPosition.getTimestamp().orElse(null));
return dt != null ? new DateTimeType(dt.withZoneSameInstant(ZoneId.systemDefault()) : UnDefType.NULL;
```